File: CascadeProjects/rudibot/test_live_connections.py
#!/usr/bin/env python3
"""
RUDIBOT Live Connection Test
Tests all critical API connections before going live
"""

import os
import sys
import json
import requests
from datetime import datetime
from dotenv import load_dotenv

# Load environment
load_dotenv()

class Colors:
    GREEN = '\033[92m'
    RED = '\033[91m'
    YELLOW = '\033[93m'
    BLUE = '\033[94m'
    ENDC = '\033[0m'
    BOLD = '\033[1m'

def test_shopify_connection():
    """Test Shopify API connection"""
    print(f"\n{Colors.BLUE}🛍️  Testing Shopify Connection{Colors.ENDC}")
    
    store_url = os.getenv('SHOPIFY_STORE_URL')
    token = os.getenv('SHOPIFY_ADMIN_TOKEN')
    api_version = os.getenv('SHOPIFY_API_VERSION', '2025-01')
    
    if not store_url or not token:
        print(f"{Colors.RED}❌ Shopify credentials not found in .env{Colors.ENDC}")
        return False
    
    if 'your-store' in store_url or 'your_real' in token:
        print(f"{Colors.YELLOW}⚠️  Shopify credentials are placeholders{Colors.ENDC}")
        return False
    
    try:
        # Test store info
        url = f"https://{store_url}/admin/api/{api_version}/shop.json"
        headers = {
            'X-Shopify-Access-Token': token,
            'Content-Type': 'application/json'
        }
        
        response = requests.get(url, headers=headers, timeout=10)
        
        if response.status_code == 200:
            shop_data = response.json().get('shop', {})
            print(f"{Colors.GREEN}✅ Shopify Store: {shop_data.get('name', 'Unknown')}{Colors.ENDC}")
            print(f"   📍 Domain: {shop_data.get('domain', 'Unknown')}")
            print(f"   💰 Currency: {shop_data.get('currency', 'Unknown')}")
            return True
        else:
            print(f"{Colors.RED}❌ Shopify API Error: {response.status_code}{Colors.ENDC}")
            print(f"   {response.text}")
            return False
            
    except Exception as e:
        print(f"{Colors.RED}❌ Shopify Connection Error: {str(e)}{Colors.ENDC}")
        return False

def test_server_health():
    """Test local server health"""
    print(f"\n{Colors.BLUE}🏥 Testing Server Health{Colors.ENDC}")
    
    try:
        response = requests.get('http://localhost:3200/api/health', timeout=5)
        
        if response.status_code == 200:
            data = response.json()
            print(f"{Colors.GREEN}✅ Server Status: {data.get('status', 'Unknown')}{Colors.ENDC}")
            print(f"   ⏱️  Uptime: {data.get('uptime', 'Unknown')}")
            print(f"   💾 Memory: {data.get('memory', 'Unknown')}")
            print(f"   🔧 Node: {data.get('node', 'Unknown')}")
            
            # Check environment
            env = data.get('env', {})
            shopify_ok = env.get('shopify1', False)
            print(f"   🛍️  Shopify Configured: {'✅' if shopify_ok else '❌'}")
            
            return True
        else:
            print(f"{Colors.RED}❌ Server Health Check Failed: {response.status_code}{Colors.ENDC}")
            return False
            
    except Exception as e:
        print(f"{Colors.RED}❌ Server Connection Error: {str(e)}{Colors.ENDC}")
        print(f"   {Colors.YELLOW}💡 Make sure server is running: npm start{Colors.ENDC}")
        return False

def test_shopify_orders():
    """Test Shopify orders endpoint"""
    print(f"\n{Colors.BLUE}📦 Testing Shopify Orders{Colors.ENDC}")
    
    store_url = os.getenv('SHOPIFY_STORE_URL')
    token = os.getenv('SHOPIFY_ADMIN_TOKEN')
    api_version = os.getenv('SHOPIFY_API_VERSION', '2025-01')
    
    if not store_url or not token:
        print(f"{Colors.RED}❌ Shopify credentials not found{Colors.ENDC}")
        return False
    
    try:
        url = f"https://{store_url}/admin/api/{api_version}/orders.json?limit=5"
        headers = {
            'X-Shopify-Access-Token': token,
            'Content-Type': 'application/json'
        }
        
        response = requests.get(url, headers=headers, timeout=10)
        
        if response.status_code == 200:
            orders = response.json().get('orders', [])
            print(f"{Colors.GREEN}✅ Orders Retrieved: {len(orders)} recent orders{Colors.ENDC}")
            
            if orders:
                latest = orders[0]
                print(f"   📋 Latest Order: #{latest.get('order_number', 'Unknown')}")
                print(f"   💰 Total: {latest.get('total_price', 'Unknown')} {latest.get('currency', '')}")
                print(f"   📅 Date: {latest.get('created_at', 'Unknown')[:10]}")
            
            return True
        else:
            print(f"{Colors.RED}❌ Orders API Error: {response.status_code}{Colors.ENDC}")
            return False
            
    except Exception as e:
        print(f"{Colors.RED}❌ Orders Error: {str(e)}{Colors.ENDC}")
        return False

def main():
    """Run all connection tests"""
    print(f"{Colors.BOLD}{Colors.BLUE}")
    print("╔══════════════════════════════════════════════════════════════╗")
    print("║           RUDIBOT LIVE CONNECTION TEST SUITE                ║")
    print("║                    Phase 1 Validation                       ║")
    print(f"╚══════════════════════════════════════════════════════════════╝{Colors.ENDC}")
    
    print(f"\n{Colors.BOLD}📅 Test Run: {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}{Colors.ENDC}")
    
    results = []
    
    # Test server first
    results.append(test_server_health())
    
    # Test Shopify connections
    results.append(test_shopify_connection())
    results.append(test_shopify_orders())
    
    # Summary
    print(f"\n{Colors.BOLD}═══════════════════════════════════════════════════════════════{Colors.ENDC}")
    print(f"{Colors.BOLD}📊 TEST SUMMARY{Colors.ENDC}")
    
    passed = sum(results)
    total = len(results)
    
    if passed == total:
        print(f"{Colors.GREEN}✅ ALL TESTS PASSED ({passed}/{total}){Colors.ENDC}")
        print(f"\n{Colors.GREEN}🎉 RUDIBOT is ready for Phase 1!{Colors.ENDC}")
        print(f"   🚀 Shopify connection is live")
        print(f"   📊 Orders and products can be fetched")
        print(f"   🔄 Ready for auto-operation setup")
    else:
        print(f"{Colors.RED}❌ SOME TESTS FAILED ({passed}/{total}){Colors.ENDC}")
        print(f"\n{Colors.YELLOW}🔧 Next steps:{Colors.ENDC}")
        
        if not results[0]:  # Server failed
            print(f"   1. Start server: npm start")
            print(f"   2. Check port 3200 is available")
        
        if not results[1]:  # Shopify connection failed
            print(f"   1. Update .env with real Shopify credentials")
            print(f"   2. Create admin token with correct scopes")
            print(f"   3. Verify store URL is correct")
        
        if not results[2]:  # Orders failed
            print(f"   1. Check admin token permissions")
            print(f"   2. Verify API version compatibility")
    
    print(f"\n{Colors.BOLD}═══════════════════════════════════════════════════════════════{Colors.ENDC}")
    
    return passed == total

if __name__ == "__main__":
    success = main()
    sys.exit(0 if success else 1)

File: CascadeProjects/rudibot/scripts/test-live-connections.js
#!/usr/bin/env node
/**
 * Live-Connection Check for RUDIBOT
 * Phase 1: Shopify Re-activation & API health
 *
 * Usage:
 *   node scripts/test-live-connections.js
 *   node scripts/test-live-connections.js --verbose
 *   node scripts/test-live-connections.js --write-env
 */
'use strict';
require('dotenv').config();

const fs = require('fs');
const path = require('path');

// ── Config ───────────────────────────────────────────────────
const CHECKS = {
  shopify: {
    name: 'Shopify Store',
    required: true,
    url: () => `https://${process.env.SHOPIFY_STORE_URL}/admin/api/${process.env.SHOPIFY_API_VERSION || '2025-01'}/shop.json`,
    headers: () => ({ 'X-Shopify-Access-Token': process.env.SHOPIFY_ADMIN_TOKEN }),
    validate: async (json) => {
      if (!json.shop) throw new Error('Missing shop object');
      return {
        store: json.shop.name,
        domain: json.shop.domain,
        plan: json.shop.plan_name,
        currency: json.shop.currency,
        timezone: json.shop.iana_timezone,
        products_count: json.shop.products_count,
        orders_count: json.shop.orders_count,
      };
    }
  },
  shopifyOrders: {
    name: 'Shopify Orders (Live)',
    required: true,
    url: () => `https://${process.env.SHOPIFY_STORE_URL}/admin/api/${process.env.SHOPIFY_API_VERSION || '2025-01'}/orders.json?status=any&limit=5`,
    headers: () => ({ 'X-Shopify-Access-Token': process.env.SHOPIFY_ADMIN_TOKEN }),
    validate: async (json) => {
      if (!Array.isArray(json.orders)) throw new Error('Missing orders array');
      const latest = json.orders[0];
      return {
        total_orders: json.orders.length,
        latest_order_id: latest?.id || null,
        latest_order_number: latest?.name || null,
        latest_order_total: latest?.total_price || null,
        latest_order_date: latest?.created_at || null,
        financial_status: latest?.financial_status || null,
      };
    }
  },
  shopifyProducts: {
    name: 'Shopify Products',
    required: false,
    url: () => `https://${process.env.SHOPIFY_STORE_URL}/admin/api/${process.env.SHOPIFY_API_VERSION || '2025-01'}/products.json?limit=5`,
    headers: () => ({ 'X-Shopify-Access-Token': process.env.SHOPIFY_ADMIN_TOKEN }),
    validate: async (json) => ({
      total_products: json.products?.length || 0,
      first_product: json.products?.[0]?.title || null,
    })
  },
  printify: {
    name: 'Printify',
    required: false,
    url: () => 'https://api.printify.com/v1/shops.json',
    headers: () => ({ 'Authorization': `Bearer ${process.env.PRINTIFY_API_KEY}` }),
    validate: async (json) => ({
      shops: json.data?.map(s => ({ id: s.id, title: s.title })) || [],
    })
  },
  paypal: {
    name: 'PayPal (OAuth)',
    required: false,
    url: () => {
      const base = process.env.PAYPAL_SANDBOX === 'true'
        ? 'https://api-m.sandbox.paypal.com'
        : 'https://api-m.paypal.com';
      return `${base}/v1/oauth2/token`;
    },
    headers: () => {
      const auth = Buffer.from(`${process.env.PAYPAL_CLIENT_ID}:${process.env.PAYPAL_CLIENT_SECRET}`).toString('base64');
      return {
        'Authorization': `Basic ${auth}`,
        'Content-Type': 'application/x-www-form-urlencoded',
      };
    },
    method: 'POST',
    body: 'grant_type=client_credentials',
    validate: async (json) => {
      if (!json.access_token) throw new Error('No access_token');
      return { token_type: json.token_type, expires_in: json.expires_in };
    }
  },
  klaviyo: {
    name: 'Klaviyo',
    required: false,
    url: () => 'https://a.klaviyo.com/api/profiles/',
    headers: () => ({
      'Authorization': `Klaviyo-API-Key ${process.env.KLAVIYO_API_KEY}`,
      'revision': '2023-02-22',
    }),
    validate: async (json) => ({ profiles_count: json.data?.length || 0 })
  },
  telegram: {
    name: 'Telegram Bot',
    required: false,
    url: () => `https://api.telegram.org/bot${process.env.TELEGRAM_BOT_TOKEN}/getMe`,
    headers: () => ({}),
    validate: async (json) => {
      if (!json.ok) throw new Error(json.description || 'Telegram API error');
      return { bot_username: json.result.username, bot_id: json.result.id };
    }
  },
  bankImportPath: {
    name: 'Bank CSV Import Path',
    required: false,
    skipHttp: true,
    validate: async () => {
      const p = process.env.BANK_IMPORT_PATH || './imports/bank';
      const exists = fs.existsSync(p);
      const files = exists ? fs.readdirSync(p).filter(f => f.endsWith('.csv')) : [];
      return { path: p, exists, csv_files: files.length, csv_names: files.slice(0, 5) };
    }
  },
  envFile: {
    name: '.env File Health',
    required: true,
    skipHttp: true,
    validate: async () => {
      const envPath = path.join(__dirname, '..', '.env');
      const exists = fs.existsSync(envPath);
      const content = exists ? fs.readFileSync(envPath, 'utf-8') : '';
      const secrets = [];
      const lines = content.split('\n');
      const criticalKeys = [
        'SHOPIFY_STORE_URL', 'SHOPIFY_ADMIN_TOKEN',
        'PAYPAL_CLIENT_ID', 'PAYPAL_CLIENT_SECRET',
        'PRINTIFY_API_KEY', 'PRINTIFY_SHOP_ID',
        'TELEGRAM_BOT_TOKEN', 'TELEGRAM_CHAT_ID',
        'ANTHROPIC_API_KEY', 'OPENAI_API_KEY',
      ];
      const foundKeys = [];
      for (const line of lines) {
        const match = line.match(/^([A-Z_]+)\s*=\s*(.+)$/);
        if (match) {
          const [, key, val] = match;
          if (criticalKeys.includes(key)) foundKeys.push(key);
          if (val.length > 4 && !val.includes('PLACEHOLDER') && !val.includes('your_')) {
            // looks like a real secret
          } else if (val.includes('PLACEHOLDER') || val.includes('your_')) {
            secrets.push(`${key} = PLACEHOLDER`);
          }
        }
      }
      const missingKeys = criticalKeys.filter(k => !foundKeys.includes(k));
      return {
        env_exists: exists,
        critical_keys_found: foundKeys.length,
        critical_keys_total: criticalKeys.length,
        missing_keys: missingKeys,
        placeholder_count: secrets.length,
        placeholders: secrets.slice(0, 5),
      };
    }
  }
};

// ── Colors ────────────────────────────────────────────────────
const C = {
  reset: '\x1b[0m',
  green: '\x1b[32m',
  red: '\x1b[31m',
  yellow: '\x1b[33m',
  cyan: '\x1b[36m',
  dim: '\x1b[2m',
  bold: '\x1b[1m',
};

function ok(msg) { return `${C.green}✅${C.reset} ${msg}`; }
function fail(msg) { return `${C.red}❌${C.reset} ${msg}`; }
function warn(msg) { return `${C.yellow}⚠️${C.reset}  ${msg}`; }
function info(msg) { return `${C.cyan}ℹ️${C.reset}  ${msg}`; }

// ── Runner ────────────────────────────────────────────────────
async function runCheck(key, config, verbose) {
  const start = Date.now();
  const result = {
    key,
    name: config.name,
    required: config.required,
    status: 'pending',
    latency_ms: 0,
    data: null,
    error: null,
  };

  try {
    let json;
    if (config.skipHttp) {
      json = await config.validate();
      result.data = json;
    } else {
      const url = config.url();
      const method = config.method || 'GET';
      const headers = config.headers();
      const body = config.body || undefined;

      const res = await fetch(url, { method, headers, body, signal: AbortSignal.timeout(15000) });
      result.latency_ms = Date.now() - start;

      const text = await res.text();
      try { json = JSON.parse(text); } catch { json = { _raw: text.slice(0, 200) }; }

      if (!res.ok) {
        throw new Error(`HTTP ${res.status}: ${json?.error || text.slice(0, 200)}`);
      }

      result.data = await config.validate(json);
    }
    result.status = 'ok';
  } catch (err) {
    result.status = 'error';
    result.error = err.message;
    result.latency_ms = Date.now() - start;
  }

  return result;
}

async function main() {
  const args = process.argv.slice(2);
  const verbose = args.includes('--verbose') || args.includes('-v');
  const writeEnv = args.includes('--write-env');

  console.log(`\n${C.bold}RUDIBOT Live Connection Check${C.reset}`);
  console.log(`${C.dim}Phase 1: Shopify Reactivation & API Health${C.reset}\n`);

  const results = [];
  for (const [key, config] of Object.entries(CHECKS)) {
    const r = await runCheck(key, config, verbose);
    results.push(r);
  }

  // ── Print Results ───────────────────────────────────────────
  let passed = 0;
  let failed = 0;
  let criticalFailed = 0;

  for (const r of results) {
    const icon = r.status === 'ok' ? ok('') : r.required ? fail('') : warn('');
    const label = r.required ? `${r.name} ${C.dim}(required)${C.reset}` : r.name;
    console.log(`${icon} ${label}`);

    if (r.status === 'ok') {
      passed++;
      if (verbose && r.data) {
        for (const [k, v] of Object.entries(r.data)) {
          if (Array.isArray(v)) {
            console.log(`   ${C.dim}${k}:${C.reset} [${v.length} items]`);
            for (const item of v.slice(0, 3)) {
              console.log(`      ${C.dim}- ${JSON.stringify(item).slice(0, 80)}${C.reset}`);
            }
          } else {
            console.log(`   ${C.dim}${k}:${C.reset} ${v}`);
          }
        }
      }
      if (r.latency_ms > 0) {
        console.log(`   ${C.dim}latency: ${r.latency_ms}ms${C.reset}`);
      }
    } else {
      failed++;
      if (r.required) criticalFailed++;
      console.log(`   ${C.red}Error:${C.reset} ${r.error}`);
    }
    console.log('');
  }

  // ── Summary ─────────────────────────────────────────────────
  console.log(`${C.bold}Summary${C.reset}`);
  console.log(`  ${ok(`Passed: ${passed}/${results.length}`)}`);
  if (failed > 0) {
    console.log(`  ${fail(`Failed: ${failed}/${results.length}`)}`);
  }
  if (criticalFailed > 0) {
    console.log(`  ${fail(`Critical failures: ${criticalFailed}`)}`);
  }

  // ── Phase 1 Gate ────────────────────────────────────────────
  const shopifyOk = results.find(r => r.key === 'shopify')?.status === 'ok';
  const ordersOk = results.find(r => r.key === 'shopifyOrders')?.status === 'ok';
  const envOk = results.find(r => r.key === 'envFile')?.status === 'ok';

  console.log(`\n${C.bold}Phase 1 Gate${C.reset}`);
  if (shopifyOk && ordersOk && envOk) {
    console.log(`  ${ok('SHOPIFY LIVE — ready for Phase 2 (Auto-Store)')}`);
    console.log(`  ${C.dim}Next: Activate order automation, Printify fulfillment, support agent${C.reset}`);
    process.exitCode = 0;
  } else {
    console.log(`  ${fail('BLOCKED — fix before Phase 2')}`);
    if (!shopifyOk) console.log(`     → Fix SHOPIFY_STORE_URL and SHOPIFY_ADMIN_TOKEN`);
    if (!ordersOk) console.log(`     → Check token scopes (needs: read_orders, read_products)`);
    if (!envOk) console.log(`     → Add missing keys to .env, replace PLACEHOLDER values`);
    process.exitCode = 1;
  }

  // ── Write .env if requested ─────────────────────────────────
  if (writeEnv && !envOk) {
    console.log(`\n${C.yellow}Writing .env template...${C.reset}`);
    const template = `# RUDIBOT Production ENV — generated by test-live-connections.js
NODE_ENV=production
PORT=3200

# Shopify (REQUIRED for Phase 1)
SHOPIFY_STORE_URL=your-store.myshopify.com
SHOPIFY_ADMIN_TOKEN=shpat_xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
SHOPIFY_API_VERSION=2025-01
SHOPIFY_WEBHOOK_SECRET=whsec_xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
SHOPIFY_LOCATION_ID=123456789

# Printify
PRINTIFY_API_KEY=your_printify_key
PRINTIFY_SHOP_ID=123456

# PayPal
PAYPAL_CLIENT_ID=your_paypal_client_id
PAYPAL_CLIENT_SECRET=your_paypal_client_secret
PAYPAL_SANDBOX=true

# Bank Import
BANK_IMPORT_PATH=./imports/bank

# Telegram
TELEGRAM_BOT_TOKEN=123456:ABC-DEF1234ghIkl-zyx57W2v1u123ew11
TELEGRAM_CHAT_ID=123456789

# AI APIs
ANTHROPIC_API_KEY=sk-ant-...
OPENAI_API_KEY=sk-...
PERPLEXITY_API_KEY=pplx-...
`;
    fs.writeFileSync(path.join(__dirname, '..', '.env.template'), template);
    console.log(`  ${ok('.env.template written to project root')}`);
  }
}

main().catch(err => {
  console.error(fail(`Fatal error: ${err.message}`));
  process.exit(1);
});
